src/components/inputs/TextInput.tsx

```diff
@@ -0,0 +1,114 @@
+    errorMessage?: string;
+    informationText?: string;
+    infoTextStyle?: TextStyle;
+    errorMessageStyle?: TextStyle;
+    errorIcon?: IconName;
+    errorIconSize?: number;
+    errorIconColor?: string;
+}
+
+export const TextInput = ({
+    style = {},
+    label,
+    value,
+    onChangeText,
+    isError,
+    errorMessage,
+    informationText,
+    infoTextStyle,
+    errorMessageStyle,
+    errorIcon = 'dlc',
+    errorIconSize = 16,
+    errorIconColor,
+    ...props
+}: Props) => {
+    const theme = useTheme();
+
+    const [focused, setFocused] = useState<boolean>(false);
+
+    const containerStyle: ViewStyle = {
+        marginBottom: theme.sw.spacing.l,
+        ...style,
+    };
+
+    const inputSyle: TextStyle = {
+        backgroundColor: theme.sw.colors.neutral[50],
+        fontSize: 16,
+        fontFamily: 'PublicSans-Regular',
+    };
+
+    const outlineStyle: ViewStyle = {
+        borderWidth: 1,
+        borderColor: isError
+            ? theme.sw.colors.error.main
+            : focused
+            ? theme.sw.colors.neutral[500]
+            : theme.sw.colors.neutral[400],
+    };
+
+    const informationTextStyle: TextStyle = {
+        marginTop: theme.sw.spacing.xs,
+        color: value.length > 0 ? theme.sw.colors.neutral[500] : theme.sw.colors.neutral[600],
+        ...infoTextStyle,
+    };
+
+    const errorContainer: ViewStyle = {
+        flexDirection: 'row',
+        marginTop: theme.sw.spacing.xs,
+        alignItems: 'center',
+    };
+
+    const errorText: TextStyle = {
+        marginLeft: theme.sw.spacing.xs,
+        color: theme.sw.colors.error.main,
+        ...errorMessageStyle,
+    };
+
+    return (
+        <View style={containerStyle}>
+            <BaseTextInput
+                {...props}
+                label={label}
+                style={inputSyle}
+                textColor={theme.sw.colors.neutral[900]}
+                mode={'outlined'}
+                onFocus={() => setFocused(true)}
+                onBlur={() => setFocused(false)}
+                error={isError}
+                outlineStyle={outlineStyle}
+                theme={{
+                    roundness: 12,
+                    colors: {
+                        error: theme.sw.colors.error.main,
+                        text: theme.sw.colors.neutral[800],
+                        onSurfaceVariant: theme.sw.colors.neutral[500],
+                        primary: theme.sw.colors.neutral[800],
+                    },
+                }}
+                value={value}
+                onChangeText={onChangeText}
+            />
+            {isError ? (
+                <View style={errorContainer}>
+                    <Icon color={errorIconColor} name={errorIcon} size={errorIconSize} />
+                    <Text style={errorText}>{errorMessage}</Text>
+                </View>
+            ) : informationText ? (
+                <Text style={informationTextStyle}>{informationText}</Text>
```

Comment:
Looks like we also need the ability to have information text with icon, and also different colors...

<img width="234" alt="Screenshot 2023-03-08 at 2 37 11 PM" src="https://user-images.githubusercontent.com/578618/223727115-5a654a18-b1aa-4840-a41c-f89e62645b17.png">

I guess the InformationText could probably be a small component by itself with is own state

Reply:
yes, i added those changes in my last commit